File: app/src/main/java/com/kota/ASFramework/PageController/ASNavigationControllerPopAnimation.java
package com.kota.ASFramework.PageController;

import android.view.animation.Animation;
import com.kota.ASFramework.Thread.ASRunner;

/* loaded from: classes.dex */
public class ASNavigationControllerPopAnimation {
  private ASViewController _source_controller;
  private ASViewController _target_controller;
  private int animation_duration = 250;
  private boolean _source_finished = false;
  private boolean _target_finished = false;
  private boolean _finished = false;
  private Animation _source_animation = null;
  private Animation _target_animation = null;

  public ASNavigationControllerPopAnimation(ASViewController aSourceController, ASViewController aTargetController) {
    this._source_controller = null;
    this._target_controller = null;
    this._source_controller = aSourceController;
    this._target_controller = aTargetController;
  }

  public void start(boolean animated) {
    if (!animated) {
      finish();
    } else {
      animate();
    }
  }

  private void animate() {
    if (this._source_controller != null && this._source_controller.getPageView() != null) {
      this._source_animation = ASPageAnimation.getFadeOutTtRightAnimation();
      this._source_animation.setDuration(this.animation_duration);this._source_animation.setAnimationListener(new Animation.AnimationListener() { // from class: com.kumi.ASFramework.PageController.ASNavigationControllerPopAnimation.1
        @Override // android.view.animation.Animation.AnimationListener
        public void onAnimationStart(Animation animation) {
        }

        @Override // android.view.animation.Animation.AnimationListener
        public void onAnimationRepeat(Animation animation) {
        }

        @Override // android.view.animation.Animation.AnimationListener
        public void onAnimationEnd(Animation animation) {
          synchronized (ASNavigationControllerPopAnimation.this) {
            ASNavigationControllerPopAnimation.this._source_controller.getPageView().onPageAnimationFinished();
            ASNavigationControllerPopAnimation.this._source_finished = true;
            ASNavigationControllerPopAnimation.this.checkFinished();
          }
        }
      });
      this._source_controller.getPageView().onPageAnimationStart();
      this._source_controller.getPageView().startAnimation(this._source_animation);
    } else {
      synchronized (this) {
        this._source_finished = true;
        checkFinished();
      }
    }
    if (this._target_controller != null && this._target_controller.getPageView() != null) {
      this._target_animation = ASPageAnimation.getFadeInFromLeftAnimation();
      this._target_animation.setDuration(this.animation_duration);
      this._target_animation.setAnimationListener(new Animation.AnimationListener() { // from class: com.kumi.ASFramework.PageController.ASNavigationControllerPopAnimation.2
        @Override // android.view.animation.Animation.AnimationListener
        public void onAnimationStart(Animation animation) {
        }

        @Override // android.view.animation.Animation.AnimationListener
        public void onAnimationRepeat(Animation animation) {
        }

        @Override // android.view.animation.Animation.AnimationListener
        public void onAnimationEnd(Animation animation) {
          synchronized (ASNavigationControllerPopAnimation.this) {
            ASNavigationControllerPopAnimation.this._target_controller.getPageView().onPageAnimationFinished();
            ASNavigationControllerPopAnimation.this._target_finished = true;
            ASNavigationControllerPopAnimation.this.checkFinished();
          }
        }
      });
      this._target_controller.getPageView().onPageAnimationStart();
      this._target_controller.getPageView().startAnimation(this._target_animation);
      return;
    }
    synchronized (this) {
      this._target_finished = true;
      checkFinished();
    }
  }

  /* JADX INFO: Access modifiers changed from: private */
  public void checkFinished() {
    if (this._source_finished && this._target_finished) {
      finish();
    }
  }

  private void finish() {
    if (!this._finished) {
      new ASRunner() { // from class: com.kumi.ASFramework.PageController.ASNavigationControllerPopAnimation.3
        @Override // com.kumi.ASFramework.Thread.ASRunner
        public void run() {
          ASNavigationControllerPopAnimation.this.onAnimationFinished();
        }
      }.runInMainThread();
      this._finished = true;
    }
  }

  public void onAnimationFinished() {
  }
}

File: app/src/main/java/com/kota/asFramework/pageController/ASNavigationControllerPopAnimation.kt
package com.kota.asFramework.pageController

import android.view.animation.Animation
import com.kota.asFramework.thread.ASCoroutine

open class ASNavigationControllerPopAnimation(
    private val sourceViewController: ASViewController?,
    private val targetViewController: ASViewController?
) {
    private var isSourceFinished = false
    private var isTargetFinished = false
    private var isFinished = false

    fun start(animated: Boolean) {
        if (!animated) {
            finish()
        } else {
            animate()
        }
    }

    private fun animate() {
        val animationDuration = 250
        if (this.sourceViewController != null && this.sourceViewController.pageView != null) {
            val fadeOutAnimation = ASPageAnimation.fadeOutTtRightAnimation
            fadeOutAnimation?.duration = animationDuration.toLong()
            fadeOutAnimation?.setAnimationListener(object : Animation.AnimationListener {
                // from class: com.kota.asFramework.pageController.ASNavigationControllerPopAnimation.1
                // android.view.animation.Animation.AnimationListener
                override fun onAnimationStart(animation: Animation?) {
                }

                // android.view.animation.Animation.AnimationListener
                override fun onAnimationRepeat(animation: Animation?) {
                }

                // android.view.animation.Animation.AnimationListener
                override fun onAnimationEnd(animation: Animation?) {
                    synchronized(this@ASNavigationControllerPopAnimation) {
                        this@ASNavigationControllerPopAnimation.sourceViewController.pageView?.onPageAnimationFinished()
                        this@ASNavigationControllerPopAnimation.isSourceFinished = true
                        this@ASNavigationControllerPopAnimation.checkFinished()
                    }
                }
            })
            this.sourceViewController.pageView?.onPageAnimationStart()
            this.sourceViewController.pageView?.startAnimation(fadeOutAnimation)
        } else {
            synchronized(this) {
                this.isSourceFinished = true
                checkFinished()
            }
        }
        if (this.targetViewController != null && this.targetViewController.pageView != null) {
            val targetAnimation = ASPageAnimation.fadeInFromLeftAnimation
            targetAnimation?.duration = animationDuration.toLong()
            targetAnimation?.setAnimationListener(object : Animation.AnimationListener {
                // from class: com.kota.asFramework.pageController.ASNavigationControllerPopAnimation.2
                // android.view.animation.Animation.AnimationListener
                override fun onAnimationStart(animation: Animation?) {
                }

                // android.view.animation.Animation.AnimationListener
                override fun onAnimationRepeat(animation: Animation?) {
                }

                // android.view.animation.Animation.AnimationListener
                override fun onAnimationEnd(animation: Animation?) {
                    synchronized(this@ASNavigationControllerPopAnimation) {
                        this@ASNavigationControllerPopAnimation.targetViewController.pageView?.onPageAnimationFinished()
                        this@ASNavigationControllerPopAnimation.isTargetFinished = true
                        this@ASNavigationControllerPopAnimation.checkFinished()
                    }
                }
            })
            this.targetViewController.pageView?.onPageAnimationStart()
            this.targetViewController.pageView?.startAnimation(targetAnimation)
            return
        }
        synchronized(this) {
            this.isTargetFinished = true
            checkFinished()
        }
    }

    private fun checkFinished() {
        if (this.isSourceFinished && this.isTargetFinished) {
            finish()
        }
    }

    private fun finish() {
        if (!this.isFinished) {
            ASCoroutine.runOnMain {
                this@ASNavigationControllerPopAnimation.onAnimationFinished()
            }
            this.isFinished = true
        }
    }

    open fun onAnimationFinished() {
    }
}